### PR TITLE
splitdrive under windows

### PR DESCRIPTION
**Pull request by augustinyiptong**
Windows paths can contain non alphanumeric characters

See: https://discourse.julialang.org/t/splitdrive-under-windows/1083

@vtjnash suggested trying a PR.

### CODE REVIEW

**Comment by KristofferC** [comment]
Note that you typically do not need to close a PR and open a new one for the same bug fix or feature. Instead you can push new commits to the same branch that the original PR is from and the PR will then incorporate those new commits.

File: test/path.jl [hunk 1]
```diff
@@ -68,6 +68,11 @@ for S in (String, GenericString)
     @test joinpath(splitdir(S(homedir()))...) == homedir()
     @test string(splitdrive(S(homedir()))...) == homedir()
 
+    if is_windows()
+        @test splitdrive("\\\\servername\\hello.world\\filename.ext") == ("\\\\servername\\hello.world","\\filename.ext")
+        @test splitdrive("\\\\servername.com\\hello.world\\filename.ext") == ("\\\\servername.com\\hello.world","\\filename.ext")
```
Comment (stevengj) [inline test/path.jl]: probably good to also have a `C:\foo\bar` test

Comment (tkelman) [inline test/path.jl]: should also wrap long lines, putting the rhs of the == on the next line would work well here

**Comment by augustinyiptong** [comment]
Hi i have made the suggested changes. Thanks,

**Comment by stevengj** [comment]
Seems like this is a bugfix that should be backported to 0.5.

**Review by stevengj — APPROVED** [review]

File: test/path.jl [hunk 2]
```diff
@@ -68,6 +68,15 @@ for S in (String, GenericString)
     @test joinpath(splitdir(S(homedir()))...) == homedir()
     @test string(splitdrive(S(homedir()))...) == homedir()
 
+    if is_windows()
+        @test splitdrive("\\\\servername\\hello.world\\filename.ext") ==
```
Comment (tkelman) [inline test/path.jl]: this is inside a loop over `S` but you aren't using `S` like all the other tests here do (testing either String or GenericString as an approximation of unicode handling)

Comment (augustinyiptong) [inline test/path.jl]: fair point. moved tests outside the loop over S

Comment (tkelman) [inline test/path.jl]: would have also worked to change the test cases to `S("...")`

**Comment by tkelman** [comment]
appveyor failure was #16555, travis 32 bit linux failure was #19803

**Pull request merged**